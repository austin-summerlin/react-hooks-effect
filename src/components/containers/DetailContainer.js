/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
import CharacterDetails from '../CharacterDetails';
import { fetchCharacterById } from '../../services/heyArnoldApi';
import { useParams } from 'react-router-dom';

function DetailContainer() {
  const [character, setCharacter] = useState({});
  const [loading, setLoading] = useState(true);

  const { id } = useParams();

  useEffect(() => {
    fetchCharacterById(id)
      .then(setCharacter)
      .finally(() => setLoading(false));
  }, [id]);

  return (
    <div>
      <h1>Character Details</h1>
      {loading ? (<h1>Loading...</h1>) : (<CharacterDetails character={character} />)}
    </div>
  );
}

export default DetailContainer;
