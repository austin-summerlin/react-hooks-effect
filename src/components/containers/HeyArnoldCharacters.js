/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
import { fetchCharacters } from '../../services/heyArnoldApi';
import CharacterList from '../CharacterList';

function HeyArnoldContainer() {
  const [loading, setLoading] = useState(true);
  const [characters, setCharacters] = useState([]);

  //useEffect replaces component did mount
  useEffect(() => {
    fetchCharacters()
      .then(setCharacters)
      .finally(() => setLoading(false));
  }, []);
  return (
    <div>
      {loading ? (<img src="https://i.imgur.com/B8SSY5p.gif" alt="loading" />) : <CharacterList characters={characters} />}
    </div>
  );
}

export default HeyArnoldContainer;





