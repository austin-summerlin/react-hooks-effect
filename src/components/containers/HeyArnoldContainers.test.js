/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import HeyArnoldCharacters from './HeyArnoldCharacters';
import DetailContainer from './DetailContainer';

describe('HeyArnoldCharacters Container', () => {
  it('renders the Hey Arnold Characters', async () => {
    render(<MemoryRouter>
      <HeyArnoldCharacters />
    </MemoryRouter>);

    const ul = await screen.findByRole('list', { name: 'character-list' });
    expect(ul).toMatchSnapshot();
  });
});

describe('Character Detail Test', () => {
  it('renders the character detail', async () => {
    render(<MemoryRouter>
      <DetailContainer name="nancy" image="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923" />
    </MemoryRouter>);

    const figure = await screen.findByRole('individual');
    expect(figure).not.toBeEmptyDOMElement();
  });
});

