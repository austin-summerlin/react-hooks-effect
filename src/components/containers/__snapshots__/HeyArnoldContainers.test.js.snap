// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeyArnoldCharacters Container renders the Hey Arnold Characters 1`] = `
<ul
  aria-label="character-list"
>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f5c"
    >
      <img
        alt="Nancy"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
      <h3>
        Nancy
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f59"
    >
      <img
        alt="Grandpa's grandfather"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      />
      <h3>
        Grandpa's grandfather
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f52"
    >
      <img
        alt="Grandpa Phil"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      />
      <h3>
        Grandpa Phil
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f58"
    >
      <img
        alt="Grandpa's father"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
      />
      <h3>
        Grandpa's father
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f70"
    >
      <img
        alt="Phoebe Heyerdahl"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
      />
      <h3>
        Phoebe Heyerdahl
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f7e"
    >
      <img
        alt="Eugene Horowitz"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
      />
      <h3>
        Eugene Horowitz
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f82"
    >
      <img
        alt="Ray"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
      />
      <h3>
        Ray
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f8b"
    >
      <img
        alt="Shelley"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
      />
      <h3>
        Shelley
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f76"
    >
      <img
        alt="Lila Sawyer"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
      />
      <h3>
        Lila Sawyer
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f7c"
    >
      <img
        alt="Uncle Stinky"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
      />
      <h3>
        Uncle Stinky
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f55"
    >
      <img
        alt="Arnie"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
      />
      <h3>
        Arnie
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f66"
    >
      <img
        alt="Olga Pataki"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
      />
      <h3>
        Olga Pataki
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f6b"
    >
      <img
        alt="Timberly Johanssen"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
      />
      <h3>
        Timberly Johanssen
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f6d"
    >
      <img
        alt="Jerry Berman"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
      />
      <h3>
        Jerry Berman
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f73"
    >
      <img
        alt="Rhonda Wellington Lloyd"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
      />
      <h3>
        Rhonda Wellington Lloyd
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f78"
    >
      <img
        alt="Stinky Peterson"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
      />
      <h3>
        Stinky Peterson
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f71"
    >
      <img
        alt="Kyo Heyerdahl"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
      />
      <h3>
        Kyo Heyerdahl
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f8e"
    >
      <img
        alt="Iggy"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
      />
      <h3>
        Iggy
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f77"
    >
      <img
        alt="Mr. Sawyer"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
      />
      <h3>
        Mr. Sawyer
      </h3>
    </a>
  </li>
  <li>
    <a
      class="link-display"
      href="/5da237699734fdcb7bef8f84"
    >
      <img
        alt="Thaddeus \\"Curly\\" Gammelthorpe"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
      />
      <h3>
        Thaddeus "Curly" Gammelthorpe
      </h3>
    </a>
  </li>
</ul>
`;
